streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableAggregateTest.java

```diff
@@ -73,46 +58,54 @@ public void testAggBasic() {
-                stringSerialzied)
+                stringSerialized)
             .aggregate(
                 MockInitializer.STRING_INIT,
                 MockAggregator.TOSTRING_ADDER,
                 MockAggregator.TOSTRING_REMOVER,
                 Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("topic1-Canonized")
                     .withValueSerde(stringSerde));
 
+        final MockProcessorSupplier<String, Object> supplier = new MockProcessorSupplier<>();
         table2.toStream().process(supplier);
 
-        driver.setUp(builder, stateDir, Serdes.String(), Serdes.String());
-
-        driver.process(topic1, "A", "1");
-        driver.flushState();
-        driver.process(topic1, "B", "2");
-        driver.flushState();
-        driver.process(topic1, "A", "3");
-        driver.flushState();
-        driver.process(topic1, "B", "4");
-        driver.flushState();
-        driver.process(topic1, "C", "5");
-        driver.flushState();
-        driver.process(topic1, "D", "6");
-        driver.flushState();
-        driver.process(topic1, "B", "7");
-        driver.flushState();
-        driver.process(topic1, "C", "8");
-        driver.flushState();
-
-        assertEquals(
-            asList(
-                "A:0+1 (ts: 0)",
-                "B:0+2 (ts: 0)",
-                "A:0+1-1+3 (ts: 0)",
-                "B:0+2-2+4 (ts: 0)",
-                "C:0+5 (ts: 0)",
-                "D:0+6 (ts: 0)",
-                "B:0+2-2+4-4+7 (ts: 0)",
-                "C:0+5-5+8 (ts: 0)"),
-            supplier.theCapturedProcessor().processed);
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                builder.build(),
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                0L)) {
+            final ConsumerRecordFactory<String, String> recordFactory =
+                new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer(), 0L, 0L);
+
+            driver.pipeInput(recordFactory.create(topic1, "A", "1"));
+            driver.pipeInput(recordFactory.create(topic1, "B", "2"));
+            driver.pipeInput(recordFactory.create(topic1, "A", "3"));
+            driver.pipeInput(recordFactory.create(topic1, "B", "4"));
+            driver.pipeInput(recordFactory.create(topic1, "C", "5"));
+            driver.pipeInput(recordFactory.create(topic1, "D", "6"));
+            driver.pipeInput(recordFactory.create(topic1, "B", "7"));
+            driver.pipeInput(recordFactory.create(topic1, "C", "8"));
+
+            assertEquals(
+                asList(
+                    "A:0+1 (ts: 0)",
+                    "B:0+2 (ts: 0)",
+                    "A:0+1-1 (ts: 0)",
```

Comment:
This seems different to the old expected result.

Reply:
Hm, I figured it's because TTD forwards every record through one at a time. Even though we flush after each change, the old code would still "coalesce" the retraction and update. Does this seem wrong to you?

Reply:
Interesting. Should be fine. Still not sure if I understand why...

> Hm, I figured it's because TTD forwards every record through one at a time.

Yes. But a single input record result in an add plus remove, so I am wondering why those are not "coalesce" in TTD, too? The caching in the stores should behave the same...

> Does this seem wrong to you?

Not necessarily. Just try to understand why TTD behaves differently.

Reply:
It seems that caching is disabled, but I am not sure why?

Reply:
@vvcephei Still would like to understand this. Why does caching not take effect any longer?

Reply:
@mjsax @vvcephei I ran the new commit locally and I think I get the difference here:

In the ToplogyTestDriver#pipeInput:

```
// Process the record ...
            task.process();
            task.maybePunctuateStreamTime();
            task.commit();
            captureOutputRecords();
```

I.e. each record would cause a commit immediately, and in this case, when processing the two records from the repartition topics, each of them will trigger the `pipeInput` once and hence commit once, i.e. the processing of the original one `pipeInput` would cause two `pipeInput` from the repartition topic, and hence commit twice, and flush twice.

While in the old `KStreamTestDriver`, we do not commit from the repartition-topic piped record, hence only result in one flush.

Reply:
I feel that generally speaking the `commit-on-every-pipeInput` of TopologyTestDriver is debatable, especially since we call `pipeInput` recursively from repartition topics, which means each of the new / old records via the repartition topic would be triggering once. Will merge this PR still as-is and we can discuss if we want to change this behavior later.

Reply:
Ah. That makes sense. Thanks!

Not sure if we want/need to change the behavior. Also, it would require a KIP imho, because people may have tests in place testing for the current behavior... Not sure if it's worth it.